core/src/main/c/share/ooo_radix.h

```diff
@@ -0,0 +1,832 @@
+                    if constexpr (n > 4) {
+                        radix_shuffle<24u>(counts[n - 4], buff2, buff1, size);
+                        if constexpr (n > 5) {
+                            radix_shuffle<32u>(counts[n - 5], buff1, buff2, size);
+                            if constexpr (n > 6) {
+                                radix_shuffle<40u>(counts[n - 6], buff2, buff1, size);
+                                if constexpr (n > 7) {
+                                    radix_shuffle<48u>(counts[n - 7], buff1, buff2, size);
+                                    radix_shuffle_clean<56u, TxnBits, TRevIdx>(
+                                            counts[n - 8], buff2, out, size,
+                                            min_value, result_format);
+                                } else {
+                                    radix_shuffle_clean<48u, TxnBits, TRevIdx>(
+                                            counts[n - 7], buff1, out, size,
+                                            min_value, result_format);
+                                }
+                            } else {
+                                radix_shuffle_clean<40u, TxnBits, TRevIdx>(
+                                        counts[n - 6], buff2, out, size, min_value, result_format);
+                            }
+                        } else {
+                            radix_shuffle_clean<32u, TxnBits, TRevIdx>(
+                                    counts[n - 5], buff1, out, size, min_value, result_format);
+                        }
+                    } else {
+                        radix_shuffle_clean<24u, TxnBits, TRevIdx>(
+                                counts[n - 4], buff2, out, size, min_value, result_format);
+                    }
+                } else {
+                    radix_shuffle_clean<16u, TxnBits, TRevIdx>(
+                            counts[n - 3], buff1, out, size, min_value, result_format);
+                }
+            } else {
+                radix_shuffle_clean<8u, TxnBits, TRevIdx>(counts[n - 2], buff2, out, size, min_value, result_format);
+            }
+        } else {
+            radix_shuffle_clean<0u, TxnBits, TRevIdx>(counts[n - 1], buff1, out, size, min_value, result_format);
+        }
+        return size;
+    }
+    return 0;
+}
+
+
+template<uint16_t TsBits, uint16_t TxnBits>
+uint64_t
+radix_copy_segments_index_asc_rev(
+        const int64_t *lag_ts, const uint64_t lag_size,
+        const index_l **segment_ts_maps,
+        const txn_info *segment_txns,
+        const uint64_t txn_count,
+        index_l *out,
+        index_l *cpy,
+        const uint32_t segment_count,
+        int64_t min_value,
+        int32_t total_row_count_bytes,
+        int64_t total_row_count,
+        uint8_t result_format,
+        uint16_t segment_bits
+) {
+    if (total_row_count_bytes != integral_type_bytes(range_bytes(total_row_count + 1))) {
+        return -100;
```

Comment:
magic?

Reply:
Yeah, negative is an error and it is traceble to this point in the code in case debugging is required, I think it's ok.